sql-parser/src/main/java/io/crate/sql/Literals.java

```diff
@@ -24,8 +24,17 @@
 
 import com.google.common.annotations.VisibleForTesting;
 
+import java.util.function.UnaryOperator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class Literals {
 
+    private static Pattern octalPattern = Pattern.compile("\\\\([0-7]{1,3})");
+    private static Pattern hexPattern = Pattern.compile("\\\\x([0-9A-F]{1,2})");
+    private static Pattern uni16Pattern = Pattern.compile("\\\\u([0-9A-F]{4})");
+    private static Pattern uni32Pattern = Pattern.compile("\\\\U([0-9A-F]{8})");
```

Comment:
Afaik the case of `U` does not define if 16 or 32bit, it could be used for both.